app/models/course.rb

```diff
@@ -263,10 +263,40 @@ def sites_not_associated_with_course
   end
 
   def has_bursary?
```

Comment:
Can we not add these as database fields and populate them based on this (or whatever the correct logic is)?